doc/releases/migration-guide-3.7.rst

```diff
@@ -150,6 +150,13 @@ Analog-to-Digital Converter (ADC)
 Bluetooth HCI
 =============
 
+ * The ``BT_HCI_VS_EXT`` Kconfig option was merged into the :kconfig:option:`BT_HCI_VS` Kconfig
+   option.
+ * The ``BT_HCI_VS_EVT`` Kconfig option was removed, since vendor event support is implicit if
+   the :kconfig:option:`BT_HCI_VS` option is enabled.
+ * The bt_read_static_addr() API was removed. This wasn't really a completely public API, but
+   since it was exposed by the public hci_driver.h header file the removal is mentioned here.
```

Comment:
It's not completely clear to me when things belong in the migration guide vs. release notes. 

These lines here don't seem like a guide to me though, and as such perhaps belong in the release notes instead?

Reply:
> It's not completely clear to me when things belong in the migration guide vs. release notes.
> 
> These lines here don't seem like a guide to me though, and as such perhaps belong in the release notes instead?

@Thalley the way I think about it is in terms of "will this break someone's application". If the answer is yes, then I go for the migration guide, otherwise the release notes. In principle these are all changes that *could* break an application, although the `bt_read_static_addr()` API is kind of borderline: even though it's in a public header, the only user of it should really just have been the host stack.

Reply:
@Thalley ah, I think I understand what you mean by "does it guide you". For `BT_HCI_VS_EXT` the guide is "start using `BT_HCI_VS` instead. For `BT_HCI_VS_EVT` and `bt_read_static_addr()` the advice would (I suppose) be "don't use them". Anyway, I don't mind moving these to the release notes as well, but my understanding was that things that may require changes to applications should go to the migration guide. @carlescufi any thoughts?

Reply:
> things that may require changes to applications should go to the migration guide

I agree with the above, that as a migration guide, an user needs to take necessary action to resolve build errors and to resolve change in runtime behaviors, if any.

Reply:
If that is the case, shouldn't the migration guide still be a subset of the release notes? 

https://docs.zephyrproject.org/latest/releases/index.html#release-notes states
> Changes that require the user to modify their own application to support the new release may be mentioned in the release notes, but the details regarding what needs to be changed are to be detailed in the release’s migration guide.

So release could be considered _what_ and migration guide could be considered _how_?

Reply:
The wording is a bit confusing. Anything that requires changing the user's code must be in the migration guide. The release notes is about changes in general, and the criteria for what to include there is a bit more loosely defined. I would place this in the migration guide only in this case.

Reply:
I will send a PR to clarify this.